File: DuDoong-Common/src/main/java/band/gosrock/common/annotation/ApiErrorCodeExample.java
package band.gosrock.common.annotation;


import band.gosrock.common.exception.BaseErrorCode;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ApiErrorCodeExample {
    Class<? extends BaseErrorCode> value();
}

File: DuDoong-Common/src/main/kotlin/band/gosrock/common/annotation/ApiErrorCodeExample.kt
package band.gosrock.common.annotation

import band.gosrock.common.exception.BaseErrorCode
import kotlin.reflect.KClass

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class ApiErrorCodeExample(
    val value: KClass<out BaseErrorCode>,
)
